src/languages/purebasic.js

```diff
@@ -0,0 +1,64 @@
+/*
+Language: PureBASIC
+Author: Tristano Ajmone <tajmone@gmail.com>
+Description: Syntax highlighting for PureBASIC (v.5). No inline ASM highlighting. First release (v.1.0), April 2016.
+Credits: I've taken inspiration from the PureBasic language file for GeSHi, created by Gustavo Julio Fiorenza (GuShH).
+Category: misc
+*/
+
+// Base deafult colors in PB IDE: background: #FFFFDF; foreground: #000000;
+
+function(hljs) {
+	var STRINGS = { // PB IDE color: #0080FF (Azure Radiance)
+		className: 'string',
+		begin: '(~)?"', end: '"',
+		illegal: '\\n',
+		relevance: 0
+	};
+	var CONSTANTS = { // PB IDE color: #924B72 (Cannon Pink)
+		//  "#" + a letter or underscore + letters, digits or underscores + (optional) "$"
+			className: 'symbol',
+			begin: '#[a-zA-Z_]\\w*\\$?',
+			relevance: 0
+	};
+	var PROCEDURES_RHS = { // PB IDE color: #006666 (Blue Stone)
+		// ")"
+		className: 'function',
+		begin: '\\)',
+		relevance: 0
+	};
+
+  return {
+		aliases: ['pb', 'pbi'],
+		keywords: {
+			keyword: // PB IDE color: #006666 (Blue Stone)
+				// The following keywords list was taken and adapted from GuShH's PureBasic language file for GeSHi...
+				'And As Break CallDebugger Case CompilerCase CompilerDefault CompilerElse CompilerEndIf CompilerEndSelect ' +
+				'CompilerError CompilerIf CompilerSelect Continue Data DataSection EndDataSection Debug DebugLevel Declare ' +
+				'DeclareCDLL DeclareDLL Default Define Dim DisableASM DisableDebugger DisableExplicit Else ElseIf EnableASM ' +
+				'EnableDebugger EnableExplicit End EndEnumeration EndIf EndImport EndInterface EndMacro EndProcedure ' +
+				'EndSelect EndStructure EndStructureUnion EndWith Enumeration Extends FakeReturn For Next ForEach ' +
+				'ForEver Global Gosub Goto If Import ImportC IncludeBinary IncludeFile IncludePath Interface Macro ' +
+				'NewList Not Or Procedure ProcedureC ProcedureCDLL ProcedureDLL ProcedureReturn Protected Prototype ' +
+				'PrototypeC Read ReDim Repeat Until Restore Return Select Shared Static Step Structure StructureUnion ' +
+				'Swap To Wend While With XIncludeFile XOr'
+		},
+		contains: [
+			// PROCEDURES_LHS | PB IDE color: #006666 (Blue Stone)
+			{
+				className: 'function',
+				begin: '[a-zA-Z_]\\w*\\s*\\(',
+				relevance: 0,
+			},
```

Comment:
Ok. I've been studying some other langs sources, and I seem to gather that what you are telling me is that I should implement in the code to capture Procedure definitions by intercepting their keyword (ie: `Procedure` and `Declare`, in PureBASIC), and have it contain the Procedure `Title` (ie: the identifier that follows these keywords, before opening brace).

I can manage that, but I have a couple of questions:

* Since I will be defining the `Procedure` and `Declare` keywords in the RE that captures their declaration, should I remove them from the main keywords list?

* As for capturing and highlighting the Procedure _calls_, what tag should these calls go by?

Hjs documentation is good, but at times a bit tight in examples (especially regarding the HTML final tags), and I have trouble with the CSS classes reference page — I don't seem to find a class for functions/procedure class.

PureBASIC employs a very simple and minimalistic highlighting style, but I would like to be able to catch all the required case in order to be able to reproduce on my custom stylesheet the exact coloring of its IDE. So I need to catch Procedure calls and tag them somehow.

Any suggestions?